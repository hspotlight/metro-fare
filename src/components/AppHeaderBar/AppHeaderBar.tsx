import React from "react";
import {
  AppBar,
  Grid,
  IconButton,
  Toolbar,
  Typography,
} from "@material-ui/core";
import MenuIcon from "@material-ui/icons/Menu";
import { useDrawerContext } from "../../contexts/DrawerProvider";
import { canShowSideMenuDrawer } from "../../config/featureToggle";

export const AppHeaderBar = () => {
  const { setSideMenu } = useDrawerContext();

  return (
    <AppBar position="static">
      <Toolbar className={"toolbar"}>
        <Grid container justify="space-between" alignItems="center">
          <img
            height="32"
            width="32"
            src="metro-fare-logo.jpg"
            alt="Metro Fare logo"
          />
          <Typography variant="h6" style={{ marginLeft: "8px", flexGrow: 1 }}>
            MetroFare
          </Typography>
          {canShowSideMenuDrawer() && (
            <IconButton edge="start" color="inherit" aria-label="menu">
              <MenuIcon onClick={() => setSideMenu(true)} />
            </IconButton>
          )}
        </Grid>
      </Toolbar>
    </AppBar>
  );
};
