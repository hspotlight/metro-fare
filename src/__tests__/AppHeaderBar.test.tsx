import React from "react";
import renderer from "react-test-renderer";
import {AppHeaderBar} from "../components/AppHeaderBar/AppHeaderBar";
import { render, cleanup, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";


describe("<AppHeaderBar /> ", () => {
  it("renders correctly", () => { //a snapshot test
    const tree = renderer.create(<AppHeaderBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders site name correctly", ()=>{
    const { queryByTestId, queryByText } = render(<AppHeaderBar />);
    expect(queryByText("MetroFare")).toBeTruthy();
  })

  it("renders drawer button correctly", ()=>{
    const { queryByTestId, queryByText } = render(<AppHeaderBar />);
    expect(queryByTestId("drawer-btn")).toBeTruthy();
  })

});
