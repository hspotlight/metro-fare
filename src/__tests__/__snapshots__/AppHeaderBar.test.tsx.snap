// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHeaderBar />  renders correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar css-1t29gy6-MuiToolbar-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-9cyib4-MuiGrid-root"
    >
      <img
        alt="Metro Fare logo"
        height="32"
        src="metro-fare-logo.jpg"
        width="32"
      />
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        style={
          Object {
            "flexGrow": 1,
            "marginLeft": "8px",
          }
        }
      >
        MetroFare
      </h6>
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1d6w9lk-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="drawer-btn"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
    </div>
  </div>
</header>
`;
